activestate.yaml

```diff
@@ -67,25 +67,44 @@ scripts:
         echo "actionlint was not found on your PATH, installing .."
         go install github.com/rhysd/actionlint/cmd/actionlint@v1.6.23
       fi
+      if { [[ "$GOOS" == "windows" ]] || [[ "$OS" == "Windows_NT" ]]; } && ! type "goversioninfo" &> /dev/null; then
+        echo "actionlint was not found on your PATH, installing .."
+        GOFLAGS="" go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+      fi
   - name: install-deps-os
     language: bash
     if: ne .OS.Name "Linux"
     description: Install OS specific deps
     value: ""
+  - name: install-deps-ci
+    language: bash
+    if: ne .Shell "cmd"
+    value: |
+      if { [[ "$GOOS" == "windows" ]] || [[ "$OS" == "Windows_NT" ]]; } && ! type "goversioninfo" &> /dev/null; then
+        echo "actionlint was not found on your PATH, installing .."
+        GOFLAGS="" go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.4.0
+      fi
   - name: preprocess
     language: bash
     description: Generates assets required by the project that aren't just specific to the build
     value: |
       set -e
       $constants.SET_ENV
       go run scripts/constants-generator/main.go ${@} -- internal/constants/generated.go
+      if [[ "$GOOS" == "windows" || "$OS" == "Windows_NT" ]]; then
+        go run scripts/versioninfo-generator/main.go version.txt cmd/state/versioninfo.json "State Tool"
+        go run scripts/versioninfo-generator/main.go version.txt cmd/state-svc/versioninfo.json "State Service"
+      fi
   - name: build
     language: bash
     description: Builds the project with the host OS as the target OS.
     value: |
       set -e
       $constants.SET_ENV
 
+      pushd cmd/state > /dev/null
```

Comment:
Please add a comment that this is to generate syso from versioninfo.txt (if that's indeed what it does).